### PR TITLE
Fix compost application rules to Minecolonies crops

### PR DESCRIPTION
**Pull request by Thodor12**
Closes #10103 

# Changes proposed in this pull request:
- Respect the max age of the crop blocks to not unnecessarily consume compost
- Fix the particles and sounds not being shown/heard when using compost
- Remove logic from farmland/crop blocks into compost item

[X] Yes I tested this before submitting it.
[ ] I also did a multiplayer test.

Review please

### CODE REVIEW

**Review by Raycoms — APPROVED** [review]

File: src/main/java/com/minecolonies/core/items/ItemCompost.java [hunk 1]
```diff
@@ -23,19 +31,52 @@ public ItemCompost(final Item.Properties properties)
...
+    {
+        BlockState state = level.getBlockState(pos);
+        if (state.getBlock() instanceof MinecoloniesCropBlock cropBlock)
+        {
+            if (!cropBlock.isMaxAge(state))
+            {
+                if (level instanceof ServerLevel serverLevel)
+                {
+                    cropBlock.attemptGrow(state, serverLevel, pos);
+                    stack.shrink(1);
```
Comment (gisellevonbingen) [inline src/main/java/com/minecolonies/core/items/ItemCompost.java:54]: please check player is creative mode.

Comment (Thodor12) [reply src/main/java/com/minecolonies/core/items/ItemCompost.java:54]: Unnecessary, vanilla bonemeal action also does not take this into account, creative players can just grab a new stack of compost anyway.

Comment (Raycoms) [reply src/main/java/com/minecolonies/core/items/ItemCompost.java:54]: afaik vanilla does

Comment (Thodor12) [reply src/main/java/com/minecolonies/core/items/ItemCompost.java:54]:
```java
    public static boolean applyBonemeal(ItemStack p_40628_, Level p_40629_, BlockPos p_40630_, Player player) {
        BlockState blockstate = p_40629_.getBlockState(p_40630_);
        int hook = ForgeEventFactory.onApplyBonemeal(player, p_40629_, p_40630_, blockstate, p_40628_);
        if (hook != 0) {
            return hook > 0;
        } else {
            if (blockstate.getBlock() instanceof BonemealableBlock) {
                BonemealableBlock bonemealableblock = (BonemealableBlock)blockstate.getBlock();
                if (bonemealableblock.isValidBonemealTarget(p_40629_, p_40630_, blockstate, p_40629_.isClientSide)) {
                    if (p_40629_ instanceof ServerLevel) {
                        if (bonemealableblock.isBonemealSuccess(p_40629_, p_40629_.random, p_40630_, blockstate)) {
                            bonemealableblock.performBonemeal((ServerLevel)p_40629_, p_40629_.random, p_40630_, blockstate);
                        }

                        p_40628_.shrink(1);
                    }

                    return true;
                }
            }

            return false;
        }
    }
```
That is the vanilla implementation, it always shrinks the input stack regardless of creative or not

Comment (Raycoms) [reply src/main/java/com/minecolonies/core/items/ItemCompost.java:54]: Odd, I remember testing stuff with bonemeal and it never ran out

Comment (Thodor12) [reply src/main/java/com/minecolonies/core/items/ItemCompost.java:54]: I can boot it up again and check, but the code seems to suggest there's no creative check

Comment (Thodor12) [reply src/main/java/com/minecolonies/core/items/ItemCompost.java:54]: Interesting, despite always hitting the stack shrink, if I'm creative the stack does not actually shrink at all.
Must be the inventory class of the player knowing not to consume anything in creative or something?
So anyhow this is a non issue.

**Review by someaddons — APPROVED** [review]

**Pull request merged**